libs/common/hexutils.hpp

```diff
@@ -34,28 +33,27 @@ namespace iroha {
    * @return - raw bytes converted string or boost::noneif provided string
    * was not a correct hex string
    */
-  inline boost::optional<std::string> hexstringToBytestring(
-      const std::string &str) {
-    if (str.empty() or str.size() % 2 != 0) {
-      return boost::none;
+  inline iroha::expected::Result<std::string, std::string>
+  hexstringToBytestringResult(const std::string &str) {
+    using namespace iroha::expected;
+    if (str.empty()) {
+      return makeError("Empty hex string.");
     }
-    std::string result(str.size() / 2, 0);
-    for (size_t i = 0; i < result.length(); ++i) {
-      std::string byte = str.substr(i * 2, 2);
-      size_t pos = 0;  // processed characters count
-      try {
-        result.at(i) =
-            static_cast<std::string::value_type>(std::stoul(byte, &pos, 16));
-      } catch (const std::invalid_argument &) {
-        return boost::none;
-      } catch (const std::out_of_range &) {
-        return boost::none;
-      }
-      if (pos != byte.size()) {
-        return boost::none;
-      }
+    std::string result;
+    result.reserve(str.size() / 2);
```

Comment:
maybe you take `reserve` for `resize`? here I only do a probabilistic space allocation, but the actual container size is set by the inserter.

Reply:
and also that the size is not enough means that the insert operation was called more than [s/2] times, which in case s%2==1 would mean that this last character was interpreted as valid hex byte. this is against the function contract, which throws an exception in this case.

Reply:
https://en.cppreference.com/w/cpp/string/basic_string/reserve
>If new_cap is greater than the current capacity(), new storage is allocated, and capacity() is made equal or greater than new_cap.

Reply:
capacity != size